Closes #122 Closes #120

sadchla-codes opened the pull request:
`quiet` argument in `spec_to_metacore()` and `select_dataset` functions will suppress optional messages, notes and warnings when it is set to TRUE.

sadchla-codes committed:
Closes #120 to ensure `quiet` argument in `spec_to_metacore()` and `select_dataset` functions is truly quiet

sadchla-codes committed:
Adding a test coverage to check that quiet=TRUE works properly

sadchla-codes committed:
Adding more test

LiamHobby commented on R/metacore.R:
```diff
@@ -391,11 +397,15 @@ metacore <- function(ds_spec = tibble(dataset = character(), structure = charact
 #'
 select_dataset <- function(.data, dataset, simplify = FALSE, quiet = FALSE) {
 
+   quiet_if_true <- function(expr) {
```
I really like this `quiet_if_true` pattern you've used. But I'm wondering if instead of defining it as a new function inside of the main function if we can instead pull it out as it's own function until utils.R? That way we can reuse and extend it without having to redefine it in each function that uses it.

LiamHobby commented on R/metacore.R:
```diff
@@ -372,7 +378,7 @@ metacore <- function(ds_spec = tibble(dataset = character(), structure = charact
       list2env(replaced, environment())
-      }
+   }
    MetaCore$new(ds_spec, ds_vars, var_spec, value_spec, derivations, codelist, supp, quiet)
```
I believe this also needs to be wrapped in `quiet_if_true`. If you look at the original example given in the issue, when `metacore()` is called directly the warnings still persist even if `quiet = TRUE`.

```suggestion
quiet_if_true(MetaCore$new(ds_spec, ds_vars, var_spec, value_spec, derivations, codelist, supp, quiet))
```

LiamHobby reviewed (changes requested):
Looks good so far but I think there are a few things that could be streamlined and one place where I think the call to suppressMessages was missed (when `metacore()` is called directly).
Thanks

sadchla-codes committed:
Merge branch '0.3.0' into Closes_122

sadchla-codes committed:
Adding `quiet_if_true()` in util.R, refactor `metacore()`, `spec_to_metacore()`, `select_dataset()`, and `define_to_metacore()` functions to incorporate the `quiet_if_true()` and added tests

sadchla-codes commented:
Hi @LiamHobby I have added the `quiet_if_true()` in the util.R program and refactor spec_to_metacore(), define_to_metacore(), `select_dataset()` and `metacore()` functions to used the `quiet_if_true()`. I also added some tests.

LiamHobby committed:
Merge branch '0.3.0' into Closes_122

LiamHobby reviewed (approved):
Checked against example code in original issue (#120) and reviewed new test cases.
